src/LHScriptX/Script.cpp

```diff
@@ -135,7 +135,14 @@ void Script::runCommand(const std::string& identifier, const std::vector<Token>&
 	}
 
 	const auto expected_parameters = command_signature->parameters;
-	const auto expected_size = expected_parameters.size();
+	uint32_t expected_size;
+	for (expected_size = 0; expected_size < expected_parameters.size(); ++expected_size)
+	{
+		if (command_signature->parameters[expected_size] == ParameterType::None)
+		{
+			break;
```

Comment:
Changing from `std::vector` to `std::array<..., 9>`  means that the list is fixed to 9 elements. So instead of looping until end of vector, we're looping 9 elements but stopping on the first None (indicating that there are no more elements).